### PR TITLE
Early release version

### PR DESCRIPTION
**Pull request by WojciechBarczynski**
In early release implements some features, that are not meant to be in the final version / are meant to be implemented in another way, but since we need VC right now, I implemented them this way.
Features added:
1. z_value option in input pad options
2. scale option in input pad options
3. white background when there is no video in part of an output frame

### CODE REVIEW

**Comment by WojciechBarczynski** [comment]
@DominikWolek Can you check it out?

File: lib/membrane/compositor_element.ex [hunk 1]
```diff
@@ -57,6 +76,9 @@ defmodule Membrane.VideoCompositor.CompositorElement do
 
     state = %{
       videos_positions: %{},
+      videos_z_values: %{},
+      videos_scales: %{},
+      pts_offsets: %{},
```
Comment (Janix4000) [inline lib/membrane/compositor_element.ex:81]: I think you should gather all of these properties into the `VideoOptions` struct

File: lib/membrane/compositor_element.ex [hunk 2]
```diff
@@ -114,13 +142,17 @@ defmodule Membrane.VideoCompositor.CompositorElement do
...
+      videos_scales: videos_scales
     } = state
 
     id = Map.get(pads_to_ids, pad)
 
     position = Map.get(videos_positions, id)
-    :ok = Wgpu.put_video(wgpu_state, id, caps, position)
+    z_value = Map.get(videos_z_values, id)
+    scale = Map.get(videos_scales, id)
+    :ok = Wgpu.put_video(wgpu_state, id, caps, position, z_value, scale)
```
Comment (Janix4000) [inline lib/membrane/compositor_element.ex:155]: Also, you can pass one struct with these options into the rustler NIF.

File: native/membrane_videocompositor/src/compositor/edge_rounder/mod.rs [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+                        alpha: wgpu::BlendComponent::OVER,
+                    }),
+                    format: wgpu::TextureFormat::Rgba8Unorm,
+                    write_mask: wgpu::ColorWrites::ALL,
+                })],
+            }),
+
+            multisample: wgpu::MultisampleState {
+                count: 1,
+                mask: !0,
```
Comment (Janix4000) [inline native/membrane_videocompositor/src/compositor/edge_rounder/mod.rs:63]: Is it a negation of `0`?

File: native/membrane_videocompositor/src/compositor/edge_rounder/mod.rs [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+
+            render_pass.set_pipeline(&self.pipeline);
+            render_pass.set_bind_group(0, src.bind_group.as_ref().unwrap(), &[]);
+            render_pass.set_bind_group(1, &self.common.sampler_bind_group, &[]);
+            render_pass.set_vertex_buffer(0, self.common.vertex_buffer.slice(..));
+            render_pass.set_index_buffer(
+                self.common.index_buffer.slice(..),
+                wgpu::IndexFormat::Uint16,
+            );
+            render_pass.draw_indexed(0..INDICES.len() as u32, 0, 0..1);
```
Comment (Janix4000) [inline native/membrane_videocompositor/src/compositor/edge_rounder/mod.rs:106]: I do not understand `0..INDICES.len() as u32` expression.
Isn't equivalent to `INDICES as u32`?

File: pipeline/lib/membrane/video_compositor/pipeline/utility/input_stream.ex [hunk 5]
```diff
@@ -5,8 +5,10 @@ defmodule Membrane.VideoCompositor.Pipeline.Utils.InputStream do
...
           caps: Membrane.RawVideo.t(),
           position: {non_neg_integer(), non_neg_integer()},
+          z_value: float(),
+          scale: float(),
           input: String.t() | Membrane.Source
         }
-  @enforce_keys [:caps, :position, :input]
-  defstruct caps: nil, position: nil, input: nil
+  @enforce_keys [:caps, :position, :scale, :input]
+  defstruct caps: nil, z_value: nil, position: nil, scale: nil, input: nil
```
Comment (Janix4000) [inline pipeline/lib/membrane/video_compositor/pipeline/utility/input_stream.ex:13]:
```suggestion
  @enforce_keys [:caps, :position, :scale, :input]
  defstruct @enforce_keys
```

**Review by Janix4000 — CHANGES_REQUESTED** [review]
I did not view the edge rounding shader.

Some minor changes

**Pull request closed without merging**